### PR TITLE
DEVOPS-2369 Node 4.10

### PR DESCRIPTION
**Pull request by nedvna**

### CODE REVIEW

File: test/docker-compose.single.yaml [hunk 1]
```diff
@@ -8,23 +8,18 @@ services:
...
-    image: nginx:stable-alpine
-    volumes:
-      - "./nginx_conf/nginx_node.conf:/etc/nginx/conf.d/default.conf"
+      - "./nginx_conf/nginx_node.conf:/etc/nginx/sites-enabled/workload.conf"
 
   pytest:
     image: dkr.wallarm.com/tests/smoke-tests:latest
     environment:
       WALLARM_API_HOST: ${WALLARM_API_HOST}
+      WALLARM_API_PRESET: ${WALLARM_API_PRESET:-eu1}
```
Comment (denisqsound) [inline test/docker-compose.single.yaml]: change to audit

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -72,14 +89,19 @@ jobs:
...
-          name: node.tar
+          name: node-${{ matrix.ARCH }}.tar
 
       - name: Load images
-        run: docker load -i node.tar
+        run: docker load -i node-${{ matrix.ARCH }}.tar
 
       - name: Run test
         env:
           USER_UUID: ${{ steps.secrets.outputs.user_uuid }}
```
Comment (denisqsound) [inline .github/workflows/ci.yaml:102]: TEST_RC_CANDIDATE: 'False'

File: test/docker-compose.single.yaml [hunk 3]
```diff
@@ -33,6 +28,18 @@ services:
       PYTEST_ARGS: ${PYTEST_ARGS}
       NODE_BASE_URL: http://node
       HOSTNAME_OLD_NODE: smoke-tests-old-node
+      ALLURE_ENVIRONMENT_ARCH: ${ALLURE_ENVIRONMENT_ARCH}
```
Comment (denisqsound) [inline test/docker-compose.single.yaml:31]:
```suggestion
      ALLURE_ENVIRONMENT_ARCH: ${ALLURE_ENVIRONMENT_ARCH}
      TEST_RC_CANDIDATE: 'True' 
```

File: test/docker-compose.split.yaml [hunk 4]
```diff
@@ -47,6 +43,18 @@ services:
       PYTEST_ARGS: ${PYTEST_ARGS}
       NODE_BASE_URL: http://node
       HOSTNAME_OLD_NODE: smoke-tests-old-node
+      ALLURE_ENVIRONMENT_ARCH: ${ALLURE_ENVIRONMENT_ARCH}
```
Comment (denisqsound) [inline test/docker-compose.split.yaml:46]:
```suggestion
      ALLURE_ENVIRONMENT_ARCH: ${ALLURE_ENVIRONMENT_ARCH}
      TEST_RC_CANDIDATE: 'True' 
```

File: test/docker-compose.split.yaml [hunk 5]
```diff
@@ -47,6 +43,19 @@ services:
...
+      TEST_RC_CANDIDATE: 'True'
```
Comment (denisqsound) [inline test/docker-compose.split.yaml]:
```suggestion
      TEST_RC: 'True'
```

File: test/docker-compose.single.yaml [hunk 6]
```diff
@@ -33,6 +28,19 @@ services:
...
+      TEST_RC_CANDIDATE: 'True'
```
Comment (denisqsound) [inline test/docker-compose.single.yaml]:
```suggestion
      TEST_RC: 'True'
```

File: test/docker-compose.single.yaml [hunk 7]
```diff
@@ -8,23 +8,18 @@ services:
...
-      - nginx
-
-  nginx:
-    image: nginx:stable-alpine
-    volumes:
-      - "./nginx_conf/nginx_node.conf:/etc/nginx/conf.d/default.conf"
+      - "./nginx_conf/nginx_node.conf:/etc/nginx/sites-enabled/workload.conf"
 
   pytest:
     image: dkr.wallarm.com/tests/smoke-tests:latest
```
Comment (denisqsound) [inline test/docker-compose.single.yaml:19]:
```suggestion
    image: dkr.wallarm.com/tests/smoke-tests:d4d46ce0e739e720788cdc849520faabeaddb0d8@sha256:99154927bba08cab891cfd62641f0fd5c55c092d980e7c225c6c5d5cfa48756b
```

File: test/docker-compose.split.yaml [hunk 8]
```diff
@@ -26,19 +26,15 @@ services:
...
       timeout: 5s
       retries: 10
 
-  nginx:
-    image: nginx:stable-alpine
-    volumes:
-      - "./nginx_conf/nginx_node.conf:/etc/nginx/conf.d/default.conf"
-
   pytest:
     image: dkr.wallarm.com/tests/smoke-tests:latest
```
Comment (denisqsound) [inline test/docker-compose.split.yaml:34]:
```suggestion
    image: dkr.wallarm.com/tests/smoke-tests:d4d46ce0e739e720788cdc849520faabeaddb0d8@sha256:99154927bba08cab891cfd62641f0fd5c55c092d980e7c225c6c5d5cfa48756b
```

**Comment by nedvna** [comment]
@denisqsound Feel free to commit to the branch directly (w/o suggestions)

File: scripts/init [hunk 9]
```diff
@@ -201,19 +152,29 @@ if [ -n "$POSTANALYTIC_ADDRESS" ] && [ -n "$POSTANALYTIC_ONLY" ]; then
...
 if [ -z "$WALLARM_API_TOKEN" ]; then
-  if [ ! -f "/etc/wallarm/private.key" ]; then
+  if [ ! -f "/opt/wallarm/etc/wallarm/private.key" ]; then
     echo "ERROR: no WALLARM_API_TOKEN and no private key in /etc/wallarm/private.key" >&2
     exit 1
   fi
 fi
 
+# Propagate env vars to supervisor
+set -a && source /opt/wallarm/env.list && set +a
```
Comment (pbaclawski) [inline scripts/init:162]: Could you please double check this?
I am getting such an error when running it:
```
root@4ac664631da1:/# set -a && source /opt/wallarm/env.list && set +a
bash: /opt/wallarm/usr/local/share/tarantool/?/init.lua: No such file or directory
bash: /opt/wallarm/usr/local/share/wallarm-tarantool/modules/?.lua: No such file or directory
bash: /opt/wallarm/usr/local/share/wallarm-tarantool/modules/?/init.lua: No such file or directory
```
which makes Tarantool quit.

Using `export $(sed -e 's/=\(.*\)/="\1"/g' /opt/wallarm/env.list | grep -v "#" | xargs)` fixes the issue.

At least this was an issue when running with `env.list` from the `4.8.6` version. Another fix is to quote the `LUA_PATH` value.

Comment (xDmitriev) [reply scripts/init:162]: @nedvna take a look please

Comment (nedvna) [reply scripts/init:162]: Hello @pbaclawski,

We patch this in the 4.10 image, achieved this way: https://github.com/wallarm/docker-wallarm-node/pull/53/files#diff-fa455f75ee8dccc747b5dccba98d3aca6eb6751da07161e057d6abe60d9af05fR24-R26

The way 4.8 stable is shipped is quite different, so this fix was not required there
Not sure how you got that error on 4.8 but you can use the fix from 4.10

Please note this PR is still WIP though

Comment (pbaclawski) [reply scripts/init:162]: Thank you, @nedvna!

I haven't seen this fix, hence the issue, with it, everything seems to work as expected. Thank you once again. :)

**Review by xDmitriev — APPROVED** [review]

**Pull request merged**